cmd/kafka-consumer/main.go

```diff
@@ -669,43 +674,42 @@ func (c *Consumer) ConsumeClaim(session sarama.ConsumerGroupSession, claim saram
 				}
 				if partition != target {
 					log.Panic("RowChangedEvent dispatched to wrong partition",
+						zap.Int32("partition", partition),
+						zap.Int64("offset", message.Offset),
 						zap.Int32("obtained", partition),
 						zap.Int32("expected", target),
-						zap.Int32("partitionNum", c.option.partitionNum),
 						zap.Any("row", row),
 					)
 				}
 
-				globalResolvedTs := atomic.LoadUint64(&c.globalResolvedTs)
 				partitionResolvedTs := atomic.LoadUint64(&sink.resolvedTs)
-				if row.CommitTs <= globalResolvedTs || row.CommitTs <= partitionResolvedTs {
-					log.Warn("RowChangedEvent fallback row, ignore it",
+				if row.CommitTs < partitionResolvedTs {
+					log.Panic("RowChangedEvent commit-ts less than the resolved ts",
+						zap.Int32("partition", partition),
+						zap.Int64("offset", message.Offset),
 						zap.Uint64("commitTs", row.CommitTs),
-						zap.Uint64("globalResolvedTs", globalResolvedTs),
 						zap.Uint64("partitionResolvedTs", partitionResolvedTs),
-						zap.Int32("partition", partition),
 						zap.Any("row", row))
-					// todo: mark the offset after the DDL is fully synced to the downstream mysql.
-					session.MarkMessage(message, "")
-					continue
-				}
-				var partitionID int64
-				if row.TableInfo.IsPartitionTable() {
-					partitionID = row.PhysicalTableID
 				}
-				tableID := c.fakeTableIDGenerator.
-					generateFakeTableID(row.TableInfo.GetSchemaName(), row.TableInfo.GetTableName(), partitionID)
-				row.TableInfo.TableName.TableID = tableID
 
+				tableID := row.PhysicalTableID
+				// simple protocol decoder should have set the table id already.
+				if c.option.protocol != config.ProtocolSimple {
+					var partitionID int64
+					if row.TableInfo.IsPartitionTable() {
+						partitionID = row.PhysicalTableID
+					}
+					tableID = c.fakeTableIDGenerator.
+						generateFakeTableID(row.TableInfo.GetSchemaName(), row.TableInfo.GetTableName(), partitionID)
+					row.TableInfo.TableName.TableID = tableID
```

Comment:
Why is this needed?

Reply:
to make other protocol, such as canal-json works, they does not encode the table id into the message

Reply:
Would it still overwrite the actual value of `TableInfo.TableName.TableID` as we saw in the log of failed tests?

Reply:
simple protocol won't hit this code snippets

Reply:
Got it.